src/halmos/__main__.py

```diff
@@ -753,6 +754,9 @@ def future_callback(future_model):
 
     # display assertion solving progress
     if not args.no_status or args.early_exit:
+        # creating a new live display would fail if the previous one was still active
+        rich.get_console().clear_live()
+
```

Comment:
could you elaborate on the race condition scenarios? my goal was to ensure only a single Status instance exists at a time. is it not the case, or is the Live instance a different thing from Status?

Reply:
update from offline discussion: it's not super clear, but it might be related to the use of context manager within generator.